infra/Pos/Network/Types.hs

```diff
@@ -142,8 +142,8 @@ topologySubscriptionWorker = go
 topologyRunKademlia :: Topology kademlia -> Maybe kademlia
 topologyRunKademlia = go
   where
-    go (TopologyRelay _ kademlia)         = Just kademlia
-    go (TopologyP2P _ _ kademlia)         = Just kademlia
+    go (TopologyRelay _ mKademlia)        = mKademlia
```

Comment:
Ah, the perils of catch-alls. Will fix.

The use of `go` just keeps the code layout cleaner.